### PR TITLE
[FIX][CORE] #580 Heartbeat causes vectortime desync 

### PR DESCRIPTION
**Pull request by srossbach**
The Heartbeat dispatcher forgot to register the listener and so the fireActivity call was also a NoOp as well.

### CODE REVIEW

File: core/src/saros/concurrent/management/HeartbeatDispatcher.java [hunk 1]
```diff
@@ -48,6 +49,7 @@ public void start() {
 
   private void dispatchHeartbeats() {
     // clientDocs should only be accessed by the main thread
+    // FIXME syncExec not asyncExec
```
Comment (m273d15) [inline core/src/saros/concurrent/management/HeartbeatDispatcher.java]: Could you add a reason? I think it is hard to verify whether an old "FIXME"s without a reason is still relevant.

File: core/src/saros/concurrent/management/HeartbeatDispatcher.java [hunk 2]
```diff
@@ -56,13 +58,15 @@ private void dispatchHeartbeats() {
...
+                              new NoOperation(), session.getLocalUser(), path);
                       fireActivity(heartbeat);
                     }));
   }
 
   @Override
   public void stop() {
+    session.removeActivityProducer(this);
     heartbeatScheduledExecutor.shutdown();
+    // FIXME wait here until everything is finished
```
Comment (m273d15) [inline core/src/saros/concurrent/management/HeartbeatDispatcher.java]: Could you add a reason?

Comment (srossbach) [reply core/src/saros/concurrent/management/HeartbeatDispatcher.java]: I thought the reason is obvious. After calling stop I would expect that the component is stopped. If you read the Javadoc and know the the behavior of asyncExec in combination with shutdown it is clear that this component can still do something (not the case anymore after the listener removal). However it is possible that a heartbeat is still broadcasted / executed in the ActivityHandler when you leave the stop method.

Comment (srossbach) [reply core/src/saros/concurrent/management/HeartbeatDispatcher.java]: I will address the FIXME in another patch. (Pushing it later today).

**Pull request merged**